### DISCUSSION TITLE
How to use ui.header with full height divs?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#2703] [Q&A] [answered]
### Question

Hi All,

I was wondering how to use the header at the top of the page together with other row/cols which can be full height of the page?
What I mean is the following:

If I use this, this will create a scroll bar on the page, because it's more than the page height
```python
ui.query('.nicegui-content').classes('p-0')

ui.header().classes('absolute h-[50px] bg-blue-2')
ui.column().classes('h-screen')
```
But if I replace the ui.header with a normal ui.row, then, we can keep it to normal page height
```python
ui.query('.nicegui-content').classes('p-0')

ui.row().classes('absolute w-screen h-[50px] bg-blue-2')
ui.column().classes('h-screen')
```

But I would like to keep the ```ui.header``` of course, because it has already so many built-in features. 
So how can I keep the ```ui.header``` as well as other full page height divs?

Can you please advise me on this please? Thanks a lot for your help

Regards,
Anindya

**Comment by falkoschindler** [comment, marked as answer]
Hi @Anindya088,

You can calculate the remaining height like this (where "100vh" is "100% of the viewport height):
```py
ui.column().classes('h-[calc(100vh-50px)]')
```

I don't like this solution too much, but I use it myself due to the lack of a better alternative. 🤷🏻‍♂️

**Reply by Anindya088** [reply]
Hi @falkoschindler , thanks a lot for this. I did not know we can calculate the height like that inside the ```class``` values :) Very useful to know this.

Thanks,
Anindya
